### PR TITLE
Chk 3529 job logic implementation

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-payment-wallet-scheduler-service/pull/4

#### List of Changes
Add job execution implementation: this implementation use Scheduled spring method to plan execution with a configurable cron for a batch that will process all payment wallets into a defined time window

#### Motivation and Context
Those modifications are needed in order to add chron executed job trigger

#### How Has This Been Tested?
Junit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/scheduler/config/properties/PaymentWalletJobConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package it.pagopa.wallet.scheduler.config.properties
+
+import java.time.Instant
+import org.springframework.boot.context.properties.ConfigurationProperties
+
+@ConfigurationProperties("paymentwalletjob.execution")
```
Comment (CianoDanilo) [inline src/main/kotlin/it/pagopa/wallet/scheduler/config/properties/PaymentWalletJobConfiguration.kt:6]:
```suggestion
@ConfigurationProperties("paymentWalletJob.execution")
```

File: src/main/kotlin/it/pagopa/wallet/scheduler/scheduledjob/PaymentWalletScheduledJob.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                )
+            )
+            .subscribe(
+                {
+                    logger.info(
+                        "Payment wallets processed successfully. Latest created date processed wallet: [{}]",
+                        it
+                    )
+                },
+                { logger.error("Exception processing payment wallets", it) },
```
Comment (EmanueleBVtech) [inline src/main/kotlin/it/pagopa/wallet/scheduler/scheduledjob/PaymentWalletScheduledJob.kt:37]: Avoid logging whole exception to limit risk of stacktrace being exposed
 
                `{ error ->
                    logger.error("Exception processing payment wallets: {}", error.message)
                },`

File: src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            .flatMap { cdcEventDispatcherService.dispatchEvent(it) }
+            .collectList()
+            .map {
+                it.last().timestamp // <- this is the wallet creation date and can be used to save
+                // it to
+                // Redis as latest processed chunk
+                // TODO: add redis checkpoint save here with the latest saved wallet date
+            }
+            .doOnError {
+                logger.error(
```
Comment (EmanueleBVtech) [inline src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt:107]: Avoid logging whole exception to limit risk of stacktrace being exposed

.doOnError { error ->
                logger.error(
                    "Error processing payment wallet chunk in time window: $startDate - $endDate. Error is: "+ error.message
                )
            }

**Review by EmanueleBVtech — APPROVED** [review]
Looks good, only minor changes requires attention

File: src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt (hunk 3 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt:107]: Same here, why we should not log stack trace? 

Why do you talk about expose stack trace here? those are application logs, stack trace is not put into any response body here.

File: src/main/kotlin/it/pagopa/wallet/scheduler/scheduledjob/PaymentWalletScheduledJob.kt (hunk 2 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/scheduler/scheduledjob/PaymentWalletScheduledJob.kt:37]: Why we should not log stack trace?

Why do you talk about expose stack trace here? those are application logs, stack trace is not put into any response body here.

File: src/main/kotlin/it/pagopa/wallet/scheduler/config/properties/PaymentWalletJobConfiguration.kt (hunk 1 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/scheduler/config/properties/PaymentWalletJobConfiguration.kt]: fixed with https://github.com/pagopa/pagopa-payment-wallet-scheduler-service/pull/6/commits/fed32a056ecc287a8708e8d4692569a9b38195f3

File: src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                                it.creationDate
+                                    .toString(), // use the wallet creation date as validation
+                            // operation timestamp
+                            validationErrorCode = it.validationErrorCode,
+                        )
+                )
+            }
+            .flatMap { cdcEventDispatcherService.dispatchEvent(it) }
+            .collectList()
+            .map {
```
Comment (CianoDanilo) [inline src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt:98]: Maybe a flatMap here to handle Redis I/O?

**Review by CianoDanilo — APPROVED** [review]

**Pull request merged**